interop/relayer.go

```diff
@@ -193,8 +205,25 @@ func (r *L2ToL2MessageRelayer) Stop(ctx context.Context) {
 	r.tasksCancel()
 }
 
-func (r *L2ToL2MessageRelayer) relayMessageWithRetry(l2tol2CDM *bindings.L2ToL2CrossDomainMessengerTransactor, transactor *bind.TransactOpts, sentMessage *L2ToL2MessageStoreEntry, maxRetries int) error {
+func (r *L2ToL2MessageRelayer) relayMessageWithRetry(l2tol2CDM *bindings.L2ToL2CrossDomainMessengerTransactor, crossL2Inbox *bindings.CrossL2Inbox, client *ethclient.Client, transactor *bind.TransactOpts, sentMessage *L2ToL2MessageStoreEntry, maxRetries int) error {
 	for attempt := 0; attempt < maxRetries; attempt++ {
+		crossL2InboxCaller, err := bindings.NewCrossL2InboxCaller(predeploys.CrossL2InboxAddr, client)
+		if err != nil {
+			return fmt.Errorf("failed to create caller: %w", err)
+		}
+		messageCheckSum, err := crossL2InboxCaller.CalculateChecksum(&bind.CallOpts{}, *sentMessage.Identifier(), crypto.Keccak256Hash(sentMessage.MessagePayload()))
```

Comment:
we could followup with this but i would guess there's some monorepo tooling to compute this client side

Reply:
yeah, I am planning to look into that for the viem actions. for supersim this doesnt matter much since rpc calls are free, but will definitely want to provide a client side solution for anything in production